pkg/reconciler/v1alpha1/trigger/trigger.go

```diff
@@ -503,9 +596,17 @@ func (r *reconciler) subscribeToBrokerChannel(ctx context.Context, t *v1alpha1.T
 	// Update Subscription if it has changed. Ignore the generation.
 	expected.Spec.DeprecatedGeneration = sub.Spec.DeprecatedGeneration
 	if !equality.Semantic.DeepDerivative(expected.Spec, sub.Spec) {
-		sub.Spec = expected.Spec
-		err = r.client.Update(ctx, sub)
+		// Given that the backing channel spec is immutable, we cannot just update the subscription.
+		// We delete it instead, and re-create it.
+		err = r.client.Delete(ctx, sub)
+		if err != nil {
+			logging.FromContext(ctx).Info("Cannot delete subscription", zap.Error(err))
+			return nil, err
+		}
+		sub = expected
```

Comment:
@Harwayne please, take a special look at this. We are not updating the spec only, but rather the whole object.
If I do sub.spec = expected.spec, the client then complaints that we cannot create something with a revisionNumber. Tried setting the revision to the empty string, but still complaints.

Reply:
I think this is the correct thing to do. We don't need anything from the now deleted object, we just want to make a new one.

Reply:
Sounds good!